modules/margaritas.sh

```diff
@@ -0,0 +1,170 @@
+#!/bin/bash
+function margaritas {
+  checkargn $# 2
+  num=$1
+  hflag=0
+  re='^[0-9]+$'
+  if [ "$2" = "half-full" ]; then
+    hflag=1
+  fi
+  if [ "$num" = "" ]; then
+    num=1
+  fi
+  if ! [[ $num =~ $re ]] ; then
+    echo "Please enter a valid number"
+    exit 1
+  fi
+  if [ $num -lt 0 ]; then
+    echo "Please enter a valid number of margaritas to print."
+    exit 1
+  fi
+  if [ $num -gt 1 ] || [ "$num" = "0" ]; then
```

Comment:
Should we handle exception when the user inputs a number that is too large or is letting it crash intentional?